qualtran/bloqs/arithmetic/multiplication.py

```diff
@@ -495,3 +495,81 @@ def _square_real_number() -> SquareRealNumber:
 
 
 _SQUARE_REAL_NUMBER_DOC = BloqDocSpec(bloq_cls=SquareRealNumber, examples=[_square_real_number])
+
+
+@frozen
+class InvertRealNumber(Bloq):
+    r"""Invert a fixed-point representation of a real number.
+
+    Implements the unitary:
+    $$
+        |a\rangle|0\rangle \rightarrow |a\rangle|1/a\rangle
+    $$
+    where $a \ge 1$.
+
+    Args:
+        bitsize: Number of bits used to represent the number.
+        num_frac: Number of fraction bits in the number.
+
+    Registers:
+        a: `bitsize`-sized input register.
+        result: `bitsize`-sized output register.
+
+        References:
+        [Quantum Algorithms and Circuits for Scientific Computing](
+        https://arxiv.org/pdf/1511.08253). Section 2.1.
+    """
+
+    bitsize: int
+    num_frac: int
+
+    def __attrs_post_init__(self):
+        if (
+            isinstance(self.num_frac, int)
+            and isinstance(self.bitsize, int)
+            and self.num_frac >= self.bitsize
+        ):
+            raise ValueError("num_frac must be < bitsize since a >= 1.")
+
+    @property
+    def signature(self):
+        return Signature(
+            [
+                Register("a", QFxp(self.bitsize, self.num_frac)),
+                Register("result", QFxp(self.bitsize, self.num_frac)),
+            ]
+        )
+
+    def pretty_name(self) -> str:
+        return "1/a"
+
+    def build_call_graph(self, ssa: 'SympySymbolAllocator') -> Set['BloqCountT']:
+        # import here to avoid circular dependency on qualtran.bloqs.arithmetic
+        from qualtran.bloqs.arithmetic.subtraction import Subtract
```

Comment:
This is suspicious...

Reply:
(As in signals some sort of larger problem (not really related to this PR)

Reply:
You were right to be suspicious. I found the better fix which was to qualify some imports in `subtraction.py`. Now we don't need the workaround.